### PR TITLE
Fix bugs found by Chromium's sfntly fuzzer

### PR DESCRIPTION
**Pull request by leizleiz**
Fixes https://crbug.com/641330
Fixes https://crbug.com/641446
Fixes https://crbug.com/641460
Fixes https://crbug.com/642300

### CODE REVIEW

File: cpp/src/sfntly/table/truetype/glyph_table.cc [hunk 1]
```diff
@@ -215,10 +215,11 @@ CALLER_ATTACH GlyphTable::Glyph*
 
   ReadableFontDataPtr sliced_data;
   sliced_data.Attach(down_cast<ReadableFontData*>(data->Slice(offset, length)));
-  if (type == GlyphType::kSimple) {
-    glyph = new SimpleGlyph(sliced_data);
-  } else {
-    glyph = new CompositeGlyph(sliced_data);
+  if (sliced_data) {
```
Comment (behdad) [inline cpp/src/sfntly/table/truetype/glyph_table.cc]: Well; doesn't this leave `glyph` uninitialized if `sliced_data` is nullptr?

File: cpp/src/sfntly/data/readable_font_data.cc [hunk 2]
```diff
@@ -59,23 +59,25 @@ void ReadableFontData::SetCheckSumRanges(const IntegerList& ranges) {
...
 int32_t ReadableFontData::ReadUByte(int32_t index) {
   int32_t b = array_->Get(BoundOffset(index));
-#if !defined (SFNTLY_NO_EXCEPTION)
   if (b < 0) {
+#if !defined (SFNTLY_NO_EXCEPTION)
     throw IndexOutOfBoundException(
         "Index attempted to be read from is out of bounds", index);
-  }
 #endif
+    return -1;
```
Comment (behdad) [inline cpp/src/sfntly/data/readable_font_data.cc]: Add `kInvalideUByte`?  I find it distasteful to add a constant for some and hardcoded magic-number for others.

**Comment by behdad** [comment]
lgtm with comments.  Thanks.

File: cpp/src/sfntly/table/truetype/glyph_table.cc (hunk 1 above)
Comment (leizleiz) [inline cpp/src/sfntly/table/truetype/glyph_table.cc]: |glyph| is a smart pointer, so it starts out in a consistent nullptr state internally, and returns a nullptr on Detach().

File: cpp/src/sfntly/data/readable_font_data.cc (hunk 2 above)
Comment (leizleiz) [inline cpp/src/sfntly/data/readable_font_data.cc]: Added |kInvalidUnsigned|.

**Pull request merged**